ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -186,122 +188,185 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{nam
+                                                                            onClick={() => {
+                                                                                this.appContext.apis.navigation.goto('.', {view: 'network'}, {replace: true});
+                                                                                services.viewPreferences.updatePreferences({appDetails: {...pref, view: 'network'}});
+                                                                            }}
+                                                                        />
+                                                                        <i
+                                                                            className={classNames('fa fa-th-list', {selected: pref.view === 'list'})}
+                                                                            title='List'
+                                                                            onClick={() => {
+                                                                                this.appContext.apis.navigation.goto('.', {view: 'list'}, {replace: true});
+                                                                                services.viewPreferences.updatePreferences({appDetails: {...pref, view: 'list'}});
+                                                                            }}
+                                                                        />
+                                                                    </div>
+                                                                </React.Fragment>
                                                             )
-                                                        }
-                                                        quickStarts={node => AppUtils.renderResourceButtons(node, application, tree, this.appContext, this.appChanged)}
-                                                    />
-                                                )) || (
-                                                    <div>
-                                                        <Filters pref={pref} tree={tree} onSetFilter={setFilter} onClearFilter={clearFilter}>
-                                                            {(filteredRes.length > 0 && (
+                                                        }}>
+                                                        <div className='application-details__application-search-bar'>
+                                                            <ViewPref>
+                                                                {viewPref => (
+                                                                    <DataLoader
+                                                                        key='applications'
+                                                                        load={() => AppUtils.handlePageVisibility(() => AppUtils.loadApplications(viewPref.projectsFilter))}>
```

Comment:
"Watching" application is pretty expensive: "stream" is going to send a lot of messages that will significantly increase browser CPU usage.

I think instead we should load the list of applications when user clicks search dropdown.